subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/DaemonBuildOptionFactory.java

```diff
@@ -66,7 +65,7 @@ public void applyTo(String value, DaemonParameters settings, Origin origin) {
             try {
                 settings.setIdleTimeout(new Integer(value));
             } catch (NumberFormatException e) {
-                throw new GradleException(String.format("Unable to parse %s property. The value should be an int but is: %s", gradleProperty, value));
+                origin.handleInvalidValue(GRADLE_PROPERTY, value, "the value should be an int");
```

Comment:
:x: At the moment we only provide the Gradle property but once we support the command line option then this logic becomes incorrect. We'll need to handle different origins like you do in `ParallelismBuildOptionFactory.java`.